sdk/eventhubs/azure-messaging-eventhubs/src/test/java/com/azure/messaging/eventhubs/EventHubClientBuilderTest.java

```diff
@@ -125,6 +140,107 @@ public void testProxyOptionsConfiguration(String proxyConfiguration, boolean exp
         Assertions.assertEquals(expectedClientCreation, clientCreated);
     }
 
+    @Test
+    public void sendAndReceiveEventByAzureNameKeyCredential() {
+        ConnectionStringProperties properties = getConnectionStringProperties();
+        String fullyQualifiedNamespace = getFullyQualifiedDomainName();
+        String sharedAccessKeyName = properties.getSharedAccessKeyName();
+        String sharedAccessKey = properties.getSharedAccessKey();
+        String eventHubName = getEventHubName();
+
+        final EventData testData = new EventData(TEST_CONTENTS.getBytes(UTF_8));
+
+        EventHubProducerAsyncClient asyncProducerClient = new EventHubClientBuilder()
+            .credential(fullyQualifiedNamespace, eventHubName,
+                new AzureNamedKeyCredential(sharedAccessKeyName, sharedAccessKey))
+            .buildAsyncProducerClient();
+        try {
+            asyncProducerClient.createBatch().subscribe(batch -> {
```

Comment:
The subscribe is not blocking. the callback actually runs on background thread. So this test would end.

```java
try {
    StepVerifier.create(asyncProducerClient.createBatch()
        .flatMap(batch -> {
            assertTrue(batch.tryAdd(testData));
            return asyncProducerClient.send(batch);
        })
        .verifyComplete();
} finally {
    asyncProducerClient.close();
}
```

Reply:
According to your comment in the new revision amendments have been made.